_sub/security/aws-backup-for-rds/main.tf

```diff
@@ -0,0 +1,84 @@
+# AWS Vault Creation
+resource "aws_backup_vault" "vault" {
+  count                     = var.deploy && var.vault_name != "" ? 1 : 0
+  name                      = var.vault_name
+  kms_key_arn               = var.deploy_kms_key ? aws_kms_key.backup[0].arn : var.kms_key_arn
+  tags                      = var.tags
+}
+
+# KMS Key for Encryption
+resource "aws_kms_key" "backup" {
+  count                = var.deploy && var.deploy_kms_key ? 1 : 0
```

Comment:
- Please remove `var.deploy` from count. We don't need to put that in the module. We can just put the count where we'll deploy the module. 
- I would probably name the resource `this`